src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchService.kt

```diff
@@ -74,10 +108,42 @@ fun BoolQueryBuilder.mustWhenPresent(query: String, value: LocalDate?): BoolQuer
   return this
 }
 
+fun BoolQueryBuilder.must(query: String, value: Any): BoolQueryBuilder {
+  this.must(QueryBuilders.matchQuery(query, value))
+  return this
+}
+
 fun BoolQueryBuilder.mustWhenTrue(predicate: () -> Boolean, query: String, value: String): BoolQueryBuilder {
   value.takeIf { predicate() }?.let {
     this.must(QueryBuilders
         .matchQuery(query, it))
   }
   return this
-}
\ No newline at end of file
+}
+
+fun BoolQueryBuilder.mustMultiMatchKeyword(value: String, vararg query: String): BoolQueryBuilder {
+  this.must().add(multiMatchQuery(value, *query)
+      .analyzer("whitespace")
+  )
+  return this
+}
+
+
+sealed class Result<out T, out M> {
+  data class NoMatch<out T>(val value: T? = null) : Result<T, Nothing>()
```

Comment:
Not worked out how to have a data subclass which has no fields, so this odd code will essentially return a Unit (or Void in old language) that will be null.